src/core/thread/mle.cpp

```diff
@@ -520,6 +520,7 @@ ThreadError Mle::SetRloc16(uint16_t aRloc16)
     }
 
     mMac.SetShortAddress(aRloc16);
+    mNetif.GetNetworkDataLocal().SendServerDataNotification();
```

Comment:
We might want `SendServerDataNotification() ` would be posted as a task, other than function call here.
In `Local::SendServerDataNotification()`, the verification conditions might still not be set yet when `SetRloc16()`, eg 
`mDeviceState = kDeviceStateRouter;` would not be set yet when `SetRloc16()` in `MleRouter::SetStateRouter()`, `kThreadError_InvalidState` would be returned. thus server data notification would not be sent out.

In fact for the rsdn border router, we might want the task be posted until `UpdateRoutes()`has been done at lease once in `HandleAdvertisement()` to ensure Leader is in its routing table. Because Router will find the neighbor only from its child set and routing table, obviously no valid neighbor would be found if no `UpdateRoutes()`.